obspy/io/mseed/core.py

```diff
@@ -785,6 +785,16 @@ def _write_mseed(stream, filename, encoding=None, reclen=None, byteorder=None,
                 trace_attr['encoding'] = 1
             elif trace.data.dtype.type == np.dtype(native_str('|S1')).type:
                 trace_attr['encoding'] = 0
+            #int64 data not supported; if possible, downcast to int32, otherwise create error message
+            elif trace.data.dtype.type == np.int64:
+                #check if data can be safely downcast to int32
+                trace_data1 = trace.data.copy().astype(np.int32)
+                if np.array_equal(trace.data, trace_data1) == True:
```

Comment:
Just make this test:

`if np.array_equal(trace.data, trace_data1):`

It's shorter and more readable and this is a function only returning True or False. Usually in these cases just relying on this test is fine.

As a side note, in cases where numbers are returned one has to be cautious as `0.0` as a valid number evaluates to `False` same as `None` while and any other valid number evaluates to `True`. This is a common source of bugs when overlooked. But here it's only `False` and `True` to distinguish.